chia/data_layer/data_store.py

```diff
@@ -982,14 +983,18 @@ async def set_validated_wallet_generation(self, tree_id: bytes32, generation: in
     async def get_kv_diff(
         self,
         tree_id: bytes32,
-        hash_1: Optional[bytes32],
-        hash_2: Optional[bytes32],
+        hash_1: bytes32,
+        hash_2: bytes32,
         *,
         lock: bool = True,
     ) -> Set[DiffData]:
         async with self.db_wrapper.locked_transaction(lock=lock):
-            old_pairs = set() if hash_1 is None else set(await self.get_keys_values(tree_id, hash_1, lock=False))
-            new_pairs = set() if hash_2 is None else set(await self.get_keys_values(tree_id, hash_2, lock=False))
+            old_pairs = set(await self.get_keys_values(tree_id, hash_1, lock=False))
+            new_pairs = set(await self.get_keys_values(tree_id, hash_2, lock=False))
+            if len(old_pairs) == 0 and hash_1 != bytes32([0] * 32):
+                return set()
+            if len(new_pairs) == 0 and hash_2 != bytes32([0] * 32):
+                return set()
```

Comment:
Should these be using the none bytes constant?

Is this check compensating for the store returning an empty set for roots we won't have?  Shouldn't that be fixed there an an error be the result instead of just propagating the empty set?

Reply:
this can happen quite frequently when history returns the wallet stored history, yet our local store is behind (waiting for the download to complete). we need a way to signal this to CW - empty set would be consistent with `get_keys_values` returning {} when we're behind in download.

Reply:
An error would be reasonable, however, as Florin says the empty set is consistent with some other places we return empty - like for a hash that is on-chain, but we don't have the data for (where an error would also be reasonable, but currently returns an empty thing)
Do we have a constant for `bytes32([0] * 32)`? That would seem a reasonable thing.

Reply:
I think that at some point we should stop claiming there is nothing for a hash when we do not know if there is or isn't.